### PR TITLE
Sonar / Kover / Codecov

### LINKED ISSUE #65: Sonar
Setup sonar iso EA.

### PR DESCRIPTION
**Pull request by bmarty**
This Pull Request configure sonar and codecov.

Sonar host static analysis result, see at https://sonarcloud.io/project/overview?id=vector-im_element-x-android
Codecov host code coverage result, computed by `kover`, see at https://app.codecov.io/github/vector-im/element-x-android

Codecov will also post comment on pull request with the result. Current code coverage is 20.20%.

Note that result for `develop` will be available once this PR get merged.

This PR also adds some badges to the README and update the CI so that it uploads data to sonar and codecov.

Closes #65

### CODE REVIEW

File: features/rageshake/src/main/kotlin/io/element/android/features/rageshake/logs/VectorFileLogger.kt [hunk 1]
```diff
@@ -37,6 +39,7 @@ import java.util.logging.Logger
 class VectorFileLogger(
     context: Context,
     // private val vectorPreferences: VectorPreferences
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO,
```
Comment (ganfra) [inline features/rageshake/src/main/kotlin/io/element/android/features/rageshake/logs/VectorFileLogger.kt:42]: Why not injecting Dispatchers like other classes?

File: features/rageshake/src/main/kotlin/io/element/android/features/rageshake/logs/VectorFileLogger.kt [hunk 2]
```diff
@@ -101,14 +104,14 @@ class VectorFileLogger(
...
-            tryOrNull { it.delete() }
+            it.safeDelete()
         }
     }
 
     @OptIn(DelicateCoroutinesApi::class)
     override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
         fileHandler ?: return
-        GlobalScope.launch(Dispatchers.IO) {
+        GlobalScope.launch(dispatcher) {
```
Comment (ganfra) [inline features/rageshake/src/main/kotlin/io/element/android/features/rageshake/logs/VectorFileLogger.kt:114]: If we inject dispatcher we can also inject the app CoroutineScope and remove GlobalScope

**Review by ganfra — APPROVED** [review]
One remark otherwise LGTM

File: features/rageshake/src/main/kotlin/io/element/android/features/rageshake/logs/VectorFileLogger.kt (hunk 1 above)
Comment (bmarty) [reply features/rageshake/src/main/kotlin/io/element/android/features/rageshake/logs/VectorFileLogger.kt:42]: The constructor is called from `TimberInitializer`, so not sure this is possible here.

**Pull request merged**